src/ffi.rs

```diff
@@ -40,15 +40,15 @@ extern "C" {
     /// Fills a buffer with random numbers.
     fn fill_random(buf: *mut u8, buf_len: u32) -> u8;
 
-    /// Asks the node to finds the notes for a specific view key, starting from
-    /// a certain height.
+    /// Asks the node to finds the notes for a specific view key.
     ///
-    /// The notes are to be serialized in sequence and written to `notes`, and
-    /// the number of notes written should be put in `notes_len`.
+    /// An implementor should allocate - see [`malloc`] - a buffer large enough
+    /// to contain the serialized notes and write them all in sequence. A
+    /// pointer to the first element of the buffer should then be written in
+    /// `notes`, while the number of bytes written should be put in `notes_len`.
     fn fetch_notes(
-        height: u64,
         vk: *const [u8; ViewKey::SIZE],
-        notes: *mut u8,
+        notes: *mut *mut u8,
```

Comment:
A pointer of a pointer sounds off. Can't we leave this as a normal pointer? Remember that pointers are `Copy` and we can move them around freely.

Reply:
After further investigation, we do need `*mut *mut u8` and not only `*mut *const u8` due to the workings of `Vec::from_raw_parts`. We could do some tricks, but it is quite unnecessary since in any case this all compiles down to an `i32` anyway.